### PR TITLE
fix the contains query so it works with jsonb

### LINKED ISSUE #231: the 'contains' parameter in datanommer.models Message.grep() function is broken
The issue is with:

https://github.com/fedora-infra/datanommer/blob/develop/datanommer.models/datanommer/models/__init__.py#L395

```
        if contains:
            query = query.filter(
                or_(*(Message._msg.like("%%%s%%" % contain) for contain in contains))
            )
```

_msg was previously just a string of the json message, but now it is called "msg" and is JSONB -- so the .like doesnt work anymore either

### PR DESCRIPTION
**Pull request by StephenCoady**
I've tested this locally and it's working - but I'd like to test it in staging if possible.

The reason is that I'm not 100% sure of the `msg` json key. I can see it locally but I don't think it's in all messages.

fixes: #231 
Signed-off-by: Stephen Coady <scoady@redhat.com>

### CODE REVIEW

**Comment by abompard** [comment]
Yeah we can't be sure the field has a `msg` key, I think newer fedora-messaging messages don't.

I found a way to extract the whole jsonb field to text for searching, but it uses the `#>>` operator, not the `json_extract_path_text()` function. If you can find a way to use it in SQLAlchemy I think it'll work. An example you can try in `psql`:
```
SELECT '{"f2":{"f3":1},"f4":{"f5":99,"f6":"foo"}}'::json #>> '{}';
```

In our case it would be `WHERE msg #>> '{}' LIKE <input>`, I think.

Also, please reference #231 in your commit message.

File: datanommer.models/datanommer/models/__init__.py [hunk 1]
```diff
@@ -430,6 +432,7 @@ def grep(
             return total, page, query
         else:
             # Execute!
+            print(query)
```
Comment (abompard) [inline datanommer.models/datanommer/models/__init__.py]: Leftover?

File: datanommer.models/datanommer/models/__init__.py [hunk 2]
```diff
@@ -394,7 +394,9 @@ def grep(
 
         if contains:
             query = query.filter(
-                or_(*(Message._msg.like("%%%s%%" % contain) for contain in contains))
+                or_(
+                    *(Message.msg.op("#>>")("{}").like(contain) for contain in contains)
```
Comment (abompard) [inline datanommer.models/datanommer/models/__init__.py]: So, we're not wrapping contain with percent signs anymore, should we restore that?

**Review by abompard — CHANGES_REQUESTED** [review]

**Review by abompard — APPROVED** [review]

**Comment by abompard** [comment]
Black is unhappy. You can add automatic pre-commit checks by running `pre-commit install` (install the `pre-commit` package if you don't have it yet, it's pretty cool).

**Comment by StephenCoady** [comment]
thanks for the reviews :+1:

**Pull request merged**